esmvalcore/preprocessor/_regrid.py

```diff
@@ -433,14 +434,17 @@ def _vertical_interpolate(cube, src_levels, levels, interpolation,
         src_levels.points, cube.shape, cube.coord_dims(src_levels))
 
     # force mask onto data as nan's
-    if np.ma.is_masked(cube.data):
-        cube.data[cube.data.mask] = np.nan
+    if da.ma.getmaskarray(cube.core_data()).any():
+        if cube.has_lazy_data():
+            cube.data = da.ma.filled(cube.core_data(), np.nan)
+        else:
+            cube.data[cube.data.mask] = np.nan
 
     # Now perform the actual vertical interpolation.
     new_data = stratify.interpolate(
```

Comment:
Does `stratify.interpolate` work with lazy data (i.e. when given a `dask` array instead of a `numpy` array)? I used it some time ago and I think it always realized the data. If that is the case I think replacing `cube.data` with `cube.core_data()` has no effect.

Reply:
it can ingest either - lazy or not - but it realizes the data inside its bellows - you are right about that! But there is improvement in memory from constructing the nan mask with lazy data and that is visible (30% improvement overall in function) - not bad for cases of big data, also the big impovement will be from the lazy implement of stratify but that's above my paygrade :grin:

Reply:
Awesome, then I am happy with it :smile:

Reply:
danke schoen :beer: